### PR TITLE
[release-1.2] 🌱 Update versions doc adding Kubernetes 1.25

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
Cherry-pick of #7194 (+ synced the versions.md in general a bit)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: docs/book/src/reference/versions.md [hunk 1]
```diff
@@ -63,17 +63,18 @@ These diagrams show the relationships between components in a Cluster API releas
...
-| Kubernetes v1.16  | ✓                    |                      |                     |                          |
-| Kubernetes v1.17  | ✓                    |                      |                     |                          |
-| Kubernetes v1.18  | ✓                    | ✓ (only workload)    | ✓ (only workload)   | ✓ (only workload)        |
-| Kubernetes v1.19  | ✓                    | ✓                    | ✓                   | ✓ (only workload v1.2)   |
-| Kubernetes v1.20  | ✓                    | ✓                    | ✓                   | ✓                        |
-| Kubernetes v1.21  | ✓                    | ✓                    | ✓                   | ✓                        |
-| Kubernetes v1.22  | ✓ (only workload)    | ✓                    | ✓                   | ✓                        |
-| Kubernetes v1.23* |                      | ✓                    | ✓                   | ✓                        |
-| Kubernetes v1.24  |                      |                      |                     | ✓                        |
+|                   | v0.3 (v1alpha3)      | v0.4 (v1alpha4) 			| v1.0 (v1beta1)      | v1.1 (v1beta1)    | v1.2 (v1beta1)       |
```
Comment (oscr) [inline docs/book/src/reference/versions.md:66]: nit: table misalignment here and below

**Comment by oscr** [comment]
Thank you for the update!

/lgtm

File: docs/book/src/reference/versions.md (hunk 1 above)
Comment (sbueringer) [reply docs/book/src/reference/versions.md]: Thx, fixed!

I also changed all columns to use the minimum with (text + 1 space padding on each side)

**Pull request merged**